### PR TITLE
Fix config loading when libxml_disable_entity_loader(true) is used

### PR DESCRIPTION
**Pull request by bcremer**
Fixes https://github.com/sebastianbergmann/phpunit/issues/4422

### CODE REVIEW

File: tests/unit/Util/Xml/ValidatorTest.php [hunk 1]
```diff
@@ -52,4 +52,23 @@ public function testDoesNotValidateInvalidXmlFile(): void
...
+                false,
+                true,
+                true
+            ),
+            (new SchemaFinder)->find(Version::series())
+        );
+
+        $this->assertTrue($result->hasValidationErrors());
+        $this->assertSame(PHP_EOL . '  Line 17:' . PHP_EOL . '  - Element \'filter\': This element is not expected.' . PHP_EOL, $result->asString());
+        $this->assertTrue(libxml_disable_entity_loader(true), 'libxml_disable_entity_loader must be set back to it\'s original value');
```
Comment (theseer) [inline tests/unit/Util/Xml/ValidatorTest.php:72]: To just "read" the state, we do not need to set "true".

Comment (bcremer) [reply tests/unit/Util/Xml/ValidatorTest.php:72]: I'm not aware of a read-only function to get the current libxml_disable_entity_loader-state. Die functions default value is `true`.

**Comment by bcremer** [comment]
```Deprecated: Function libxml_disable_entity_loader() is deprecated in /home/runner/work/phpunit/phpunit/src/Util/Xml/Validator.php on line 25

File: tests/unit/Util/Xml/ValidatorTest.php (hunk 1 above)
Comment (theseer) [reply tests/unit/Util/Xml/ValidatorTest.php:72]: At least according to the manual, the parameter is optional. So I at least assume that would be for "reading" the current state. I of course didn't try that ;)

**Comment by bcremer** [comment]
The PHP 8.0 runs are reporting the function call as deprecated: 

```
Deprecated: Function libxml_disable_entity_loader() is deprecated in /home/runner/work/phpunit/phpunit/src/Util/Xml/Validator.php on line 25
```

Please advise how to proceed.

**Comment by sebastianbergmann** [comment]
> Deprecated: Function libxml_disable_entity_loader() is deprecated in /home/runner/work/phpunit/phpunit/src/Util/Xml/Validator.php on line 25

This is due to http://git.php.net/?p=php-src.git;a=commitdiff;h=e0fa48f69dd14b52c8f1b2904ac7bd30472849a8.

The way I understand it, any call to `libxml_disable_entity_loader()` on PHP 8 can/should be skipped.

**Comment by theseer** [comment]
> The PHP 8.0 runs are reporting the function call as deprecated:
> 
> ```
> Deprecated: Function libxml_disable_entity_loader() is deprecated in /home/runner/work/phpunit/phpunit/src/Util/Xml/Validator.php on line 25
> ```
> 
> Please advise how to proceed.

I guess this will be getting a bit messy. If we care, we could add a condition to only call this function when PHP < 8.0 is in use and only if the current state is not what we want it to be.

If run under PHP 8, we'd have to eat the deprecation warning.

**Comment by theseer** [comment]
> The way I understand it, any call to `libxml_disable_entity_loader()` on PHP 8 can/should be skipped.


If we merely skip it the "bug" will be back, won't it? Or will PHPUnit also complain about it when it's used in a test suite?

**Comment by sebastianbergmann** [comment]
My assumption that any code, for instance @bcremer's test code and/or tested code, does not call `libxml_disable_entity_loader()` when used on PHP 8 was obviously wrong. Looks like we have to live with the deprecation warning for as long as PHPUnit supports a version of PHP that still has `libxml_disable_entity_loader()`. :-/

File: tests/unit/Util/Xml/ValidatorTest.php (hunk 1 above)
Comment (bcremer) [reply tests/unit/Util/Xml/ValidatorTest.php:72]: Unfortunately not, see: https://3v4l.org/NJprJ
 
```php
var_dump(libxml_disable_entity_loader(false)); // init with false
var_dump(libxml_disable_entity_loader()); // "read" value
var_dump(libxml_disable_entity_loader()); // value was modified
```

Comment (theseer) [reply tests/unit/Util/Xml/ValidatorTest.php:72]: \o/ Okay, never mind. *sigh*

**Comment by sebastianbergmann** [comment]
Thank you for your contribution. I appreciate the time you invested in preparing this pull request. However, I have decided not to merge it.

**Pull request closed without merging**